library-best-practices/JLBP-17.md

```diff
@@ -0,0 +1,34 @@
+[JLBP-17] Coordinate Major Version Adoption
+-------------------------------------------
+
+It is clearly undesirable for a dependency tree to contain conflicts.
+Unfortunately, as soon as any library in a dependency tree introduces a breaking
+change, consumers can't use the latest versions of those libraries together. To
+support users who try to generally use the latest versions of their
+dependencies, breaking changes need to be propagated as quickly as possible so
+that they can stay up to date.
+
+Perform the rollout in this manner:
+
+1. Decide whether to break the feature as an atomic change or to perform the
+   change in two passes, deprecate then delete.
+   - If the feature is used by stable code in other libraries, you must use two
+     passes (as per [JLBP-7](JLBP-7.md)).
+   - If the code does not appear to be used in other libraries, an in-place
+     breakage may be ok.
+   - If the feature is used by unstable code or stable code having little usage
+     itself, prefer two passes if possible, with in-place breakage only if two
+     passes are judged to be too costly.
+2. Make sure that consuming libraries are prepared for the breakage.
+   - In the case of in-place breakage, have PRs prepared and provisionally
```

Comment:
I'm still not comfortable with this. I would personally not approve a change that didn't pass CI, provisionally or otherwise.

Reply:
Well maybe "approval" is the wrong word, but the point is that the author has addressed your concerns with the code. The intent is to make the final approval quick, and not requiring a bunch of back & forth to address issues that can be caught even if CI fails. I will try to capture that somehow.